lib/recommendation-server.js

```diff
@@ -82,17 +89,36 @@ RecommendationServer.prototype = {
       req.send();
     }
   },
+
+  getResultClass: function(data) {
+    let types = Object.keys(data.enhancements);
+    if (types.indexOf('tld') > -1) {
+      return this.resultClasses.tld;
+    } else if (types.indexOf('wikipedia') > -1) {
+      return this.resultClasses.wikipedia;
+    }
+    return null;
+  },
+
   search: function(payload) {
+    let server = this;
     if (!payload.query || payload.query.length > this.maxChars) {
       return;
     }
-    this._search(payload.query).then(data => {
-      this.events.publish('recommendation', data);
+    server._search(payload.query).then(data => {
+      let resultClass = server.getResultClass(data);
```

Comment:
Since we're inside a bound function, you don't need to bind `getResultClass` in the constructor. `this` should work just fine in here.